python/lvmdrp/utils/metadata.py

```diff
@@ -0,0 +1,828 @@
+        observatory for which a metadata store will be created
+    """
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    logger.info(f"creating metadata store '{metadata_path}'")
+    store = h5py.File(metadata_path, mode="w")
+
+    # add observatory group if needed
+    if observatory not in store:
+        store.create_group(f"{observatory}/raw")
+        store.create_group(f"{observatory}/master")
+
+    store.close()
+
+
+def _del_store():
+    """delete the entire HDF store if exists"""
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    if os.path.isfile(metadata_path):
+        logger.info(f"removing metadata store '{metadata_path}'")
+        os.remove(metadata_path)
+    else:
+        logger.warning(f"no '{metadata_path}' store found")
+
+
+def _load_store(observatory, mode="r"):
+    """return the metadata store given a observatory
+
+    Parameters
+    ----------
+    observatory : str
+        metadata observatory from which a store will be loaded
+    mode : str, optional
+        the mode in which to open the HDF5 store, by default 'r'
+
+    Returns
+    -------
+    h5py.Group
+        the metadata store for the given observatory
+    """
+    metadata_path = os.path.join(access.base_dir, "metadata.hdf5")
+
+    logger.info(f"loading metadata store from '{metadata_path}'")
+    store = h5py.File(metadata_path, mode=mode)
+
+    return store[observatory]
+
+
+def extract_metadata(mjd, frames_paths):
+    """return dataframe with metadata extracted from given frames list
+
+    this function will extract metadata from FITS headers given a list of
+    frames.
+
+    Parameters
+    ----------
+    mjd : int
+        MJD from which the metadata will be extracted
+    frames_paths : list_like
+        list of frames paths in the local mirror of SAS
+
+    Returns
+    -------
+    pandas.DataFrame
+        dataframe containing the extracted metadata
+    """
+    new_metadata = {}
+    # extract metadata
+    nframes = len(frames_paths)
+    logger.info(f"extracting metadata from {nframes} frames")
+    iterator = tqdm(
+        enumerate(frames_paths),
+        total=nframes,
+        desc=f"extracting metadata from MJD = {mjd}",
+        ascii=True,
+        unit="frame",
+    )
+    for i, frame_path in iterator:
+        header = fits.getheader(frame_path, ext=0)
+        new_metadata[i] = [
+            "n" if header.get("OBSERVAT") != "LCO" else "s",
+            header.get("IMAGETYP"),
+            header.get("CCD"),
+            header.get("EXPOSURE"),
+            header.get("EXPTIME"),
+            header.get("NEON", "OFF") == "ON",
+            header.get("HGNE", "OFF") == "ON",
+            header.get("KRYPTON", "OFF") == "ON",
+            header.get("XENON", "OFF") == "ON",
+            header.get("ARGON", "OFF") == "ON",
+            header.get("LDLS", "OFF") == "ON",
+            header.get("QUARTZ", "OFF") == "ON",
+            header.get("QUALITY", QualityFlag(0)),
```

Comment:
The `QUALITY` flag in the raw frame header only refers to whether the raw exposure was a test exposure, is bad, or is good for use.  This is different than the reduction quality status of the exposure, which I think is was the `QualityFlag` refers to.  If you're planning on updating this column throughout the reduction then it's probably better this is a brand new column, e.g. `DRPQUAL`, which needs to get updated and written out to the reduced file headers.   I think we want both columns in this table.

Reply:
I added a new bitmask class `lvmdrp.utils.bitmask.RawFrameQuality`:

```python
class RawFrameQuality(BaseBitmask):
    GOOD = auto()  # bit whether a raw frame is good for reduction
    TEST = auto()  # bit whether a raw frame is for instrument testing purposes
    BAD = auto()  # bit whether a raw frame is bad for reduction
```

This will be stored in the metadata tables as the `drpqual` column.

Reply:
I think the `RawFrameQuality` should map to the `QUALITY` header, but I'm not sure we want to overwrite the original keyword.  The `QualityFlag` is what should get mapped to the new DRPQUAL header keyword.

Reply:
Yes, that's how those bitmask flags will map. I don't think we should be overwriting header keywords coming from the original (raw) frames.